corehq/apps/app_manager/templates/app_manager/partials/form_workflow.html

```diff
@@ -11,12 +11,82 @@
+                        value: workflow,
+                        event: { change: onWorkflowChange }
+                        "></select>
+                </div>
+
+                <div class="form-links-container" data-bind="
+                    visible: showFormLinkUI
+                    ">
+
+                    <!-- Container for Form Link UI. Will be populated with existing form links and where new
+                        form links will be added -->
+                    <div class="form-links" data-bind="
+                        template: {
+                            name: 'form-link-template',
+                            foreach: formLinks,
+                            as: 'formLink'
+                        }
+                        ">
+                    </div>
+                    <button class="btn btn-primary" data-bind="
+                        event: {
+                            click: onAddFormLink
+                        }
+                        ">
+                        {% trans "Add link" %}
+                    </button>
+                    <input type="hidden" class="workflow-change-trigger" />
+                </div>
             {% else %}
-                <span data-bind="text: workflow_display"></span>
+                <span data-bind="text: workflowDisplay()"></span>
             {% endif %}
         </div>
     </div>
+
+    <script type="text/html" id="form-link-template">
+        <div class="control-group">
+            <textarea name="form_links_xpath_expressions"
+                      placeholder="XPath Expression"
+                      rows="1"
+                      data-bind="
+                        text: formLink.xpath,
+                        value: formLink.xpath
+            "></textarea>
+            &nbsp;
+            <i class="icon-arrow-right"></i>
+            &nbsp;
+            <select name="form_links_form_ids" data-bind="
+                options: $parent.forms,
+                optionsText: 'name',
+                optionsValue: 'uniqueId',
+                optionsCaption: $parent.displayUnknownForm.call($parent, formLink),
+                value: formLink.formId
+                "></select>
+            <button class="btn btn-danger" data-bind="
+                event: { click: $parent.onDestroyFormLink.bind($parent) }
+                ">
+                <i class="icon-trash"></i>
+            </button>
+            <div class="alert alert-error" data-bind="
+                visible: formLink.errors().length
+                ">
+                <ul data-bind="
+                    template: {
+                        name: 'form-link-error-template',
+                        foreach: formLink.errors(),
+                        as: 'error'
+                    }
+                "></ul>
+            </div>
+
+        </div>
+    </script>
+    <script type="text/html" id="form-link-error-template">
+        <li data-bind="text: error"></li>
+    </script>
```

Comment:
Since this is just one line we probably don't need a template here and can just do a `foreach`